src/arviz_plots/plots/psensedistplot.py

```diff
@@ -0,0 +1,292 @@
+        stats_kwargs.setdefault("density", {"density": True})
+
+    if alphas is None:
+        alphas = (0.8, 1.25)
+
+    # Here we are generating new datasets for the prior and likelihood
+    # by resampling the original dataset with the power scale weights
+    # Instead we could have weighted KDEs/ecdfs/etc
+    ds_prior = new_ds(dt, "log_prior", alphas)
+    ds_likelihood = new_ds(dt, "log_likelihood", alphas)
+
+    plot_collection.coords = {"column": "prior"}
+    plot_dist(
+        ds_prior,
+        var_names=var_names,
+        filter_vars=filter_vars,
+        group=group,
+        coords=coords,
+        sample_dims=sample_dims,
+        kind=kind,
+        point_estimate=point_estimate,
+        ci_kind=ci_kind,
+        ci_prob=ci_prob,
+        plot_collection=plot_collection,
+        labeller=labeller,
+        aes_map=aes_map,
+        plot_kwargs=plot_kwargs,
+        stats_kwargs=stats_kwargs,
+    )
+    plot_collection.coords = None
+
+    plot_collection.coords = {"column": "likelihood"}
+    plot_dist(
+        ds_likelihood,
+        var_names=var_names,
+        filter_vars=filter_vars,
+        group=group,
+        coords=coords,
+        sample_dims=sample_dims,
+        kind=kind,
+        point_estimate=point_estimate,
+        ci_kind=ci_kind,
+        ci_prob=ci_prob,
+        plot_collection=plot_collection,
+        labeller=labeller,
+        aes_map=aes_map,
+        plot_kwargs=plot_kwargs,
+        stats_kwargs=stats_kwargs,
+    )
+    plot_collection.coords = None
+
+    # Overwrite the title from plot_dist to include the prior and likelihood labels
+    title_kwargs = copy(plot_kwargs.get("title", {}))
+    if title_kwargs is not False:
+        _, title_aes, title_ignore = filter_aes(plot_collection, aes_map, "title", sample_dims)
+        if "color" not in title_aes:
+            title_kwargs.setdefault("color", "black")
+
+        plot_collection.map(
+            labelled_title,
+            "title",
+            ignore_aes=title_ignore,
+            subset_info=True,
+            labeller=labeller,
+            **title_kwargs,
+        )
+
+    return plot_collection
+
+
+def new_ds(dt, group, alphas):
+    """Resample the dataset with the power scale weights."""
+    lower_w, upper_w = _get_power_scale_weights(dt, alphas, group=group)
+    lower_w = lower_w.values.flatten()
+    upper_w = upper_w.values.flatten()
+    s_size = len(lower_w)
+
+    resampled = [extract(dt, group="posterior").drop("chain")]
+    for weights in (lower_w, upper_w):
+        resampled.append(
+            extract(
+                dt,
+                group="posterior",
+                num_samples=s_size,
+                weights=weights,
+                random_seed=42,
+                resampling_method="stratified",
+            ).drop("chain")
+        )
+
+    # Middle chain is the reference chain (alpha == 1)
+    return concat(resampled, dim="chain").rename({"sample": "draw"}).sel(chain=[1, 0, 2])
```

Comment:
```suggestion
    resampled = [
        extract(
            dt,
            group="posterior",
            num_samples=s_size,
            weights=weights,
            random_seed=42,
            resampling_method="stratified",
        ).drop_indexes(["sample", "chain", "draw"])
        for weights in (lower_w, upper_w)
    ]
    resampled.insert(1, extract(dt, group="posterior").drop_indexes(["sample", "chain", "draw"]))

    return concat(resampled, dim="alpha").assign_coords(alpha=[alphas[0], 1, alphas[1]])
```

This should generate a dataset with dimensions `alpha` (to overlay over) and `sample` (to reduce) plus whatever dimensions were there already. Consequently, when we generate the legend with `plot_collection.legend("alpha")` it will title the legend `alpha` and use the correct values in the legend entries automatically.